.circleci/config.yml

```diff
@@ -0,0 +1,102 @@
+version: 2
+
+jobs:
+
+  buildImage:
+    docker:
+      - image: web3f/ci-commons:v2.4.0
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          command: |
+            /scripts/build-image.sh web3f/polkadot-registrar-challenger .
+
+  publishImage:
+    docker:
+      - image: web3f/ci-commons:v2.4.0
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          command: |
+            /scripts/publish-image.sh web3f/polkadot-registrar-challenger
+
+  publishChart:
+    docker:
+      - image: web3f/ci-commons:v2.4.0
+    steps:
+      - checkout
+      - run:
+          command: |
+            /scripts/publish-chart.sh
+
+  integrationTests:
+    docker:
+      - image: web3f/ci-commons:v2.4.0
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          description: run integration tests
+          command: |
+            /scripts/integration-tests.sh     
+            
+  deploy:
+      docker:
+        - image: web3f/ci-commons:v2.4.0
+      steps:
+        - checkout
+        - run:
+            environment:
+              GCP_REGION: europe-west3
+              HELM_ENV: production
+            command: |
+              /scripts/deploy.sh -c engineering          
+
+workflows:
+  version: 2
+  test_and_deploy:
+    jobs:   
+      - buildImage:
+          context: dockerhub-bot
+          filters:
+            tags:
+              only: /.*/
+      - integrationTests:
+          filters:
+            tags:
+              only: /.*/   
+          requires:
+            - buildImage
+      - publishImage:
+          context: dockerhub-bot
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
```

Comment:
What's going on here?

Reply:
the circleci/config.yml define the pipeline behavior and his jobs:

This specific job will occur only wen a SEMANTIC VERSION compliant tag is created

The regex is to match for the semantic version tag
Semantic versioning: https://semver.org/